opensearch_py_ml/ml_commons/IAMRoleHelper.py

```diff
@@ -0,0 +1,301 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import json
+
+import boto3
+import requests
+from botocore.exceptions import ClientError
+
+
+class IAMRoleHelper:
+    """
+    Helper class for managing IAM roles and their interactions with OpenSearch.
+    """
+
+    def __init__(
+        self,
+        region,
+        opensearch_domain_url=None,
+        opensearch_domain_username=None,
+        opensearch_domain_password=None,
+        aws_user_name=None,
+        aws_role_name=None,
+        opensearch_domain_arn=None,
+    ):
+        """
+        Initialize the IAMRoleHelper with AWS and OpenSearch configurations.
+
+        :param region: AWS region.
+        :param opensearch_domain_url: URL of the OpenSearch domain.
+        :param opensearch_domain_username: Username for OpenSearch domain authentication.
+        :param opensearch_domain_password: Password for OpenSearch domain authentication.
+        :param aws_user_name: AWS IAM user name.
+        :param aws_role_name: AWS IAM role name.
+        :param opensearch_domain_arn: ARN of the OpenSearch domain.
+        """
+        self.region = region
```

Comment:
let's re-evaluate which class variables are necessary here. I think most of them are unnecessary here.